### PR TITLE
Annotate manifests for single-node-developer cluster profile

### PR DESCRIPTION
**Pull request by guillaumerose**
This implements phase 1 of openshift/enhancements#482

### CODE REVIEW

File: manifests/0000_31_cluster-baremetal-operator_06_deployment.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ metadata:
   annotations:
     exclude.release.openshift.io/internal-openshift-hosted: "true"
     include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
```
Comment (asalkeld) [inline manifests/0000_31_cluster-baremetal-operator_06_deployment.yaml:7]: I suspect "make manifests" is going to overwite this. maybe add it here https://github.com/openshift/cluster-baremetal-operator/blob/master/config/profiles/default/kustomization.yaml#L2

Comment (guillaumerose) [reply manifests/0000_31_cluster-baremetal-operator_06_deployment.yaml:7]: done, thanks!

**Comment by sadasu** [comment]
Seems to be hitting the same well-known 4 e2e-agnostic test failures.

**Review by asalkeld — COMMENTED** [review]
/lgtm

**Pull request merged**